tests/e2e/scenarios/node-remove.go

```diff
@@ -0,0 +1,212 @@
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scenarios
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+	storageframework "k8s.io/kubernetes/test/e2e/storage/framework"
+
+	"github.com/dell/csi-baremetal-e2e-tests/e2e/common"
+)
+
+// DefineNodeRemovalTestSuite defines custom csi-baremetal node removal test
+func DefineNodeRemovalTestSuite(driver *baremetalDriver) {
+	ginkgo.Context("Baremetal-csi node remove test", func() {
+		defineNodeRemovalTest(driver)
+	})
+}
+
+func defineNodeRemovalTest(driver *baremetalDriver) {
+	ginkgo.BeforeEach(skipIfNotAllTests)
+
+	var (
+		pod           *corev1.Pod
+		pvc           *corev1.PersistentVolumeClaim
+		k8sSC         *storagev1.StorageClass
+		executor      = common.GetExecutor()
+		driverCleanup func()
+		ns            string
+		taintNodeName string
+		ctx           context.Context
+		f             = framework.NewDefaultFramework("node-remove-test")
+	)
+
+	init := func() {
+		var (
+			perTestConf *storageframework.PerTestConfig
+			err         error
+		)
+		ns = f.Namespace.Name
+		ctx = context.Background()
+		perTestConf, driverCleanup = driver.PrepareTest(f)
+		k8sSC = driver.GetDynamicProvisionStorageClass(perTestConf, "xfs")
+		k8sSC, err = f.ClientSet.StorageV1().StorageClasses().Create(context.TODO(), k8sSC, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+	}
+
+	cleanup := func() {
+		e2elog.Logf("Starting cleanup for test NodeRemoval")
+
+		if taintNodeName != "" {
+			_, _, err := executor.RunCmd(fmt.Sprintf("docker start %s", taintNodeName))
+			framework.ExpectNoError(err)
+		}
+		common.CleanupAfterCustomTest(f, driverCleanup, []*corev1.Pod{pod}, []*corev1.PersistentVolumeClaim{pvc})
+	}
+
+	ginkgo.It("CSI node resources should be deleted after node removal", func() {
+		init()
+		defer cleanup()
+
+		var err error
+		// create pvc
+		pvc, err = f.ClientSet.CoreV1().PersistentVolumeClaims(ns).Create(context.TODO(),
+			constructPVC(ns, driver.GetClaimSize(), k8sSC.Name, pvcName),
+			metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		// create pod with pvc
+		pod, err = common.CreatePod(f.ClientSet, ns, nil, []*corev1.PersistentVolumeClaim{pvc},
+			false, "sleep 3600")
+		framework.ExpectNoError(err)
+
+		e2elog.Logf("Pod %s with PVC %s created.", pod.Name, pvc.Name)
+
+		taint := corev1.Taint{
+			Key:    "node.dell.com/drain",
+			Value:  "drain:NoSchedule",
```

Comment:
Shouldn't value be `drain` only? - https://github.com/dell/csi-baremetal-operator/blob/f9346b4e8bf50213733a8a83557bfa3358495fef/pkg/nodeoperations/controller.go#L52